newdle/client/src/components/NewdleTitle.js

```diff
@@ -51,6 +52,7 @@ export default function NewdleTitle({
                     disabled={finished}
                   >
                     <Icon name="calendar plus outline" />
+                    <UnloadPrompt />
```

Comment:
This component has an `active` prop which is required - and it should only be set to `true` when there are changes where you do want the prompt to occur (ie if there are any changed that would actually be lost when changing pages).

It may also be better to move this to `AnswerPage` since it's only needed there. And you already have `participantAnswersChanged` in there to check whether the unload prompt should be active or not. Best to check for `comment !== participant.comment` as well so starting to write a comment also activates the unload prompt.

Reply:
I was working on it, but I think that there are some problems. The first time you use it (with non previous answers), const ```participantAnswersChanged``` is always false because there were no previous answers, this const is not the "real time" value, but the stored one in the previous answer.
Also is not possible to access ```comment``` const ```Cannot read property 'comment' of null``` since there are not previous comments.
In this case this piece of code:
```
  const submitDisabled =
    submitting ||
    !canSubmit ||
    (participantCode && !participant) ||
    (participantHasAnswers && !participantAnswersChanged && comment === participant.comment);
````
works the first time because ```comment``` is never evaluated (since it is the last part it never reach it), just changing the order ```(comment === participant.comment && participantHasAnswers && !participantAnswersChanged);``` the code breaks with the same error (```Cannot read property 'comment' of null```).
For these reasons I think that the first time we have to choose between prompt the alert in all cases or allow to switch and submit without any alert. The following answers it is possible to do what we want, but as I said, I think that the first one is impossible.

Reply:
Like I said, move it into the AnswerPage component ;)

And I mean we use the logic to enable the submit button, so I don't think it's impossible.... `participantAnswersChanged` should work in any case, because if you look at the selector the initial answers are an empty dict if there's no participant.